superpixel_classification/SuperpixelClassification/SuperpixelClassificationTensorflow.py

```diff
@@ -87,6 +97,29 @@ def predictLabelsForItemDetails(self, batchSize, ds, item, model, prog):
         catWeights = tf.nn.softmax(predictions)
         return catWeights, predictions
 
+    def findOptimalBatchSize(self, model, ds) -> int:
+        # Find an optimal batch_size
+        maximum_batchSize: int = len(ds)
+        batchSize: int = 2
+        # We are using a value greater than 0.0 for add_seconds so that small imprecise
+        # timings for small batch sizes don't accidentally trip the time check.
+        add_seconds: float = 0.5
```

Comment:
This is too large on my system -- it never invokes the time restriction.  I'd use a much smaller value (maybe 0.05)?

Reply:
Thank you for sleuthing that out, though I'm puzzled... it works well for my example.  I think that you are saying that processing one batch for a given batch size never takes more than 0.5 second longer than twice the time for single batch that is half that size.  I think that means either that bigger is always better or, if not, that bigger is only barely worse; which isn't what you are seeing so I am confused.

What is your dataset size, is this for training or prediction, what run times are you seeing?

Reply:
I was testing on a two image set, one with ~11,000 superpixels and the other with ~18,000 superpixels.  

Looking at the time to run the test batches for predication, I see (in seconds), on the first image, I see:
batchsize 2 0.23432278633117676 
batchsize 4 0.16736316680908203 
batchsize 8 0.0986783504486084 
batchsize 16 0.11266183853149414
batchsize 32 0.11878609657287598 
batchsize 64 0.11466693878173828
batchsize 128 0.13788604736328125
batchsize 256 0.5580830574035645 
I'd have to find a previous debug or re-add the debug statement to look beyond that.  But, here we want to stop at 128 but 0.13 * 2 + 0.5 is not less than 0.55.

This is on a system with a decent NVidia GPU.

Reply:
Okay, I've changed `add_seconds` to `0.05`.  Let's see how that works.